graphene_django/utils/utils.py

```diff
@@ -38,17 +38,24 @@ def camelize(data):
 
 
 def get_reverse_fields(model, local_field_names):
-    for name, attr in model.__dict__.items():
-        # Don't duplicate any local fields
-        if name in local_field_names:
-            continue
-
-        # "rel" for FK and M2M relations and "related" for O2O Relations
-        related = getattr(attr, "rel", None) or getattr(attr, "related", None)
-        if isinstance(related, models.ManyToOneRel):
-            yield (name, related)
-        elif isinstance(related, models.ManyToManyRel) and not related.symmetrical:
-            yield (name, related)
+    model_ancestry = [model]
+
+    for base in model.__bases__:
+        if is_valid_django_model(base):
+            model_ancestry.append(base)
+
+    for _model in model_ancestry:
+        for name, attr in _model.__dict__.items():
+            # Don't duplicate any local fields
+            if name in local_field_names:
+                continue
+
+            # "rel" for FK and M2M relations and "related" for O2O Relations
+            related = getattr(attr, "rel", None) or getattr(attr, "related", None)
+            if isinstance(related, models.ManyToOneRel):
+                yield (name, related)
+            elif isinstance(related, models.ManyToManyRel) and not related.symmetrical:
```

Comment:
@TomsOverBaghdad this being inside a loop for base classes will break relations, I think its safer to isolate getting base fields in a different method as in this case `ManyToManyRel` is two sided, in the case of `A` it is `ManyToManyRel.related` while in the case of B(A) it is `ManyToManyRel.field`.

Try having the m2m field in the base class just like the example you mentioned in the comment, not a reverse relation

Reply:
Something like this:

```
elif isinstance(related, models.ManyToManyRel) and not related.symmetrical:
    if _model is not model:
        yield (name, getattr(attr, "field", None))
        continue
```

But my preference would still be to get base models relations and reverse relations in a different method

Reply:
Heard! Good catch @firaskafri ! Let me know what you think of the changes I pushed. I separated out getting local fields and included the models ancestry similar to the reverse fields.